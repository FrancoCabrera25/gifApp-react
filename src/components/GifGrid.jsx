import { useEffect, useState } from 'react';
import {useFetchGifs} from '../hooks/useFetchGifs';
import {GifGridItem} from '../components/GifGridItem';
export const GifGrid = ({ category }) => {

  const {images, isLoading} = useFetchGifs(category);
  // const apiKey = '8avblfQHd0GqPCICPHAOw5ZYrrvhw0uI';
  // const resp   = await fetch(`http://api.giphy.com/v1/gifs/search?api_key=${ apiKey }`);

  return (
    <>
     <h3> {category}</h3>
     {
      isLoading  && (<h2>Cargando....</h2>) 
     }
      <div className='card-grid'>
        {images.map((img) => (
          <GifGridItem key={img.id} image={img}/>
        ))}
      </div>
    </>
  );
};
