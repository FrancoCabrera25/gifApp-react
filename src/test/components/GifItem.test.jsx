import { render, screen } from '@testing-library/react';
import { GifGridItem } from '../../components/GifGridItem';

describe('test gifItem', () => {
  const title = 'Valorant';
  const url = 'https://valorant/valorant.jpg';
  test('should match snapshot', () => {
    const { container } = render(<GifGridItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test('should show url y alt ', () => {
    render(<GifGridItem title={title} url={url} />);

    const { src, alt } = screen.getByRole('img');

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('should show title ', () => {
    render(<GifGridItem title={title} url={url} />);
    expect(screen.getByAltText(title)).toBeTruthy();

  });
  
});
