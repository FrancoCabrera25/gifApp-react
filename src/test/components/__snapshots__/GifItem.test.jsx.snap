// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test gifItem should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Valorant"
      src="https://valorant/valorant.jpg"
    />
    <p>
      Valorant
    </p>
  </div>
</div>
`;
